Index: Makefile
===================================================================
RCS file: /cvs/ports/cad/lepton-eda/Makefile,v
retrieving revision 1.3
diff -u -p -r1.3 Makefile
--- Makefile	27 Mar 2022 10:26:39 -0000	1.3
+++ Makefile	12 Jul 2022 18:02:01 -0000
@@ -1,17 +1,16 @@
 COMMENT =	lepton electronic design automation suite
 
-V =		1.9.17-20211219
+V =		1.9.18-20220529
 DISTNAME =	lepton-eda-${V:C/-.*//}
-REVISION =	0
 
-SHARED_LIBS +=	lepton               0.0      # 10.0
-SHARED_LIBS +=	leptongui            0.0      # 4.0
+SHARED_LIBS +=	lepton               1.0      # 11.0
+SHARED_LIBS +=	leptongui            1.0      # 5.0
 
 CATEGORIES =	cad
 
 HOMEPAGE =	https://github.com/lepton-eda/lepton-eda
 
-MAINTAINER =	Dmitry Pryakhin <graahnul.grom@gmail.com>
+MAINTAINER =	Dmitry Pryakhin <graahnul.grom@ya.ru>
 
 # GPLv2+
 PERMIT_PACKAGE =	Yes
@@ -44,7 +43,8 @@ LIB_DEPENDS =	devel/gettext,-runtime \
 
 CONFIGURE_STYLE =	gnu
 CONFIGURE_ARGS +=	--disable-attrib \
-			--disable-update-xdg-database
+			--disable-update-xdg-database \
+			--enable-guild
 
 CONFIGURE_ENV =	GROFF=false \
 		GUILE=guile2.2 \
@@ -52,17 +52,16 @@ CONFIGURE_ENV =	GROFF=false \
 		MAKEINFO=gmakeinfo \
 		M4=/usr/bin/m4
 
-# XXX make sure guile2 headers are picked up first (over guile1);               
-# cannot use CPPFLAGS/CFLAGS since they get appended last                       
-CONFIGURE_ENV +=	"CC=${CC} -I${LOCALBASE}/include/guile/2.2"             
+# XXX make sure guile2 headers are picked up first (over guile1);
+# cannot use CPPFLAGS/CFLAGS since they get appended last
+CONFIGURE_ENV +=	"CC=${CC} -I${LOCALBASE}/include/guile/2.2"
 CONFIGURE_ENV +=	"CXX=${CXX} -I${LOCALBASE}/include/guile/2.2"
 
+# suppress tons of warnings
+# "compilation of /usr/local/bin/guild2.2 failed"
+MAKE_ENV +=	"GUILE_AUTO_COMPILE=0"
+
 post-install:
-	# would be nice to precompile the guile files otherwise
-	# they're done per-user at startup. it should be similar
-	# to this, but it needs more to get it to work:
-	# cd ${WRKSRC} && ${SETENV} ${MAKE_ENV} ${MAKE_PROGRAM} \
-	#     prefix=${PREFIX} precompile
 	mv ${PREFIX}/share/doc/lepton-eda/examples \
 	    ${PREFIX}/share/examples/lepton-eda
 
Index: distinfo
===================================================================
RCS file: /cvs/ports/cad/lepton-eda/distinfo,v
retrieving revision 1.1.1.1
diff -u -p -r1.1.1.1 distinfo
--- distinfo	18 Mar 2022 09:42:36 -0000	1.1.1.1
+++ distinfo	12 Jul 2022 18:02:01 -0000
@@ -1,2 +1,2 @@
-SHA256 (lepton-eda-1.9.17.tar.gz) = lOneKeJUcw6jOX/3iv9BDWOJ3xip/vGhzxHHNAbtsS8=
-SIZE (lepton-eda-1.9.17.tar.gz) = 9454425
+SHA256 (lepton-eda-1.9.18.tar.gz) = X9yNuosNR1Jf3gYWQZeOnKdxzJLld29Sn9XYsPGWYYI=
+SIZE (lepton-eda-1.9.18.tar.gz) = 9459680
Index: pkg/PLIST
===================================================================
RCS file: /cvs/ports/cad/lepton-eda/pkg/PLIST,v
retrieving revision 1.1.1.1
diff -u -p -r1.1.1.1 PLIST
--- pkg/PLIST	18 Mar 2022 09:42:36 -0000	1.1.1.1
+++ pkg/PLIST	12 Jul 2022 18:02:07 -0000
@@ -41,6 +41,7 @@ include/liblepton/export.h
 include/liblepton/fill.h
 include/liblepton/forward.h
 include/liblepton/funcs.h
+include/liblepton/glib_compat.h
 include/liblepton/globals.h
 include/liblepton/liblepton.h
 include/liblepton/libleptonguile.h
@@ -109,8 +110,6 @@ share/doc/lepton-eda/lepton-manual.html/
 share/doc/lepton-eda/lepton-manual.html/7400-1-hidden.png
 share/doc/lepton-eda/lepton-manual.html/7400-1.png
 share/doc/lepton-eda/lepton-manual.html/Add-Component-widget.html
-share/doc/lepton-eda/lepton-manual.html/Add-attributes.html
-share/doc/lepton-eda/lepton-manual.html/Adding-attributes-to-your-symbol.html
 share/doc/lepton-eda/lepton-manual.html/Allegro-backend.html
 share/doc/lepton-eda/lepton-manual.html/Attribute-compatibility.html
 share/doc/lepton-eda/lepton-manual.html/Attributes.html
@@ -186,6 +185,7 @@ share/doc/lepton-eda/lepton-manual.html/
 share/doc/lepton-eda/lepton-manual.html/PADS-backend.html
 share/doc/lepton-eda/lepton-manual.html/Path-data.html
 share/doc/lepton-eda/lepton-manual.html/Printing-and-exporting.html
+share/doc/lepton-eda/lepton-manual.html/Program-Index.html
 share/doc/lepton-eda/lepton-manual.html/Quitting-the-program.html
 share/doc/lepton-eda/lepton-manual.html/Related-tools.html
 share/doc/lepton-eda/lepton-manual.html/Resources.html
@@ -934,6 +934,165 @@ share/lepton-eda/bitmap/gschem-text.xpm
 share/lepton-eda/bitmap/gschem-top2bottom.png
 share/lepton-eda/bitmap/gschem-undo.xpm
 share/lepton-eda/bitmap/gschem-warning.png
+share/lepton-eda/ccache/
+share/lepton-eda/ccache/auto-refdes.go
+share/lepton-eda/ccache/auto-uref.go
+share/lepton-eda/ccache/color-map.go
+share/lepton-eda/ccache/conf/
+share/lepton-eda/ccache/conf/schematic/
+share/lepton-eda/ccache/conf/schematic/attribs.go
+share/lepton-eda/ccache/conf/schematic/deprecated.go
+share/lepton-eda/ccache/conf/schematic/keys.go
+share/lepton-eda/ccache/conf/schematic/menu.go
+share/lepton-eda/ccache/conf/schematic/stroke.go
+share/lepton-eda/ccache/geda/
+share/lepton-eda/ccache/geda-deprecated-config.go
+share/lepton-eda/ccache/geda.go
+share/lepton-eda/ccache/geda/attrib.go
+share/lepton-eda/ccache/geda/config.go
+share/lepton-eda/ccache/geda/deprecated.go
+share/lepton-eda/ccache/geda/library.go
+share/lepton-eda/ccache/geda/log.go
+share/lepton-eda/ccache/geda/object.go
+share/lepton-eda/ccache/geda/os.go
+share/lepton-eda/ccache/geda/page.go
+share/lepton-eda/ccache/gschem/
+share/lepton-eda/ccache/gschem/action.go
+share/lepton-eda/ccache/gschem/attrib.go
+share/lepton-eda/ccache/gschem/builtins.go
+share/lepton-eda/ccache/gschem/deprecated.go
+share/lepton-eda/ccache/gschem/gschemdoc.go
+share/lepton-eda/ccache/gschem/hook.go
+share/lepton-eda/ccache/gschem/keymap.go
+share/lepton-eda/ccache/gschem/selection.go
+share/lepton-eda/ccache/gschem/symbol/
+share/lepton-eda/ccache/gschem/symbol/check.go
+share/lepton-eda/ccache/gschem/util.go
+share/lepton-eda/ccache/gschem/window.go
+share/lepton-eda/ccache/lepton/
+share/lepton-eda/ccache/lepton/attrib.go
+share/lepton-eda/ccache/lepton/autoplace.go
+share/lepton-eda/ccache/lepton/color-map.go
+share/lepton-eda/ccache/lepton/config.go
+share/lepton-eda/ccache/lepton/eval.go
+share/lepton-eda/ccache/lepton/ffi/
+share/lepton-eda/ccache/lepton/ffi.go
+share/lepton-eda/ccache/lepton/ffi/lib.go
+share/lepton-eda/ccache/lepton/file-system.go
+share/lepton-eda/ccache/lepton/gerror.go
+share/lepton-eda/ccache/lepton/gettext.go
+share/lepton-eda/ccache/lepton/legacy-config/
+share/lepton-eda/ccache/lepton/legacy-config.go
+share/lepton-eda/ccache/lepton/legacy-config/keylist.go
+share/lepton-eda/ccache/lepton/library/
+share/lepton-eda/ccache/lepton/library.go
+share/lepton-eda/ccache/lepton/library/component.go
+share/lepton-eda/ccache/lepton/log-rotate.go
+share/lepton-eda/ccache/lepton/log.go
+share/lepton-eda/ccache/lepton/m4.go
+share/lepton-eda/ccache/lepton/object/
+share/lepton-eda/ccache/lepton/object.go
+share/lepton-eda/ccache/lepton/object/foreign.go
+share/lepton-eda/ccache/lepton/object/text.go
+share/lepton-eda/ccache/lepton/object/type.go
+share/lepton-eda/ccache/lepton/option.go
+share/lepton-eda/ccache/lepton/os.go
+share/lepton-eda/ccache/lepton/page/
+share/lepton-eda/ccache/lepton/page.go
+share/lepton-eda/ccache/lepton/page/foreign.go
+share/lepton-eda/ccache/lepton/rc.go
+share/lepton-eda/ccache/lepton/repl.go
+share/lepton-eda/ccache/lepton/srfi-37.go
+share/lepton-eda/ccache/lepton/toplevel.go
+share/lepton-eda/ccache/lepton/version.go
+share/lepton-eda/ccache/netlist/
+share/lepton-eda/ccache/netlist.go
+share/lepton-eda/ccache/netlist/attrib/
+share/lepton-eda/ccache/netlist/attrib/compare.go
+share/lepton-eda/ccache/netlist/attrib/refdes.go
+share/lepton-eda/ccache/netlist/backend-getopt.go
+share/lepton-eda/ccache/netlist/config.go
+share/lepton-eda/ccache/netlist/deprecated.go
+share/lepton-eda/ccache/netlist/duplicate.go
+share/lepton-eda/ccache/netlist/error.go
+share/lepton-eda/ccache/netlist/hierarchy.go
+share/lepton-eda/ccache/netlist/mode.go
+share/lepton-eda/ccache/netlist/net.go
+share/lepton-eda/ccache/netlist/option.go
+share/lepton-eda/ccache/netlist/package-pin.go
+share/lepton-eda/ccache/netlist/package.go
+share/lepton-eda/ccache/netlist/partlist/
+share/lepton-eda/ccache/netlist/partlist.go
+share/lepton-eda/ccache/netlist/partlist/common.go
+share/lepton-eda/ccache/netlist/port.go
+share/lepton-eda/ccache/netlist/schematic/
+share/lepton-eda/ccache/netlist/schematic-component.go
+share/lepton-eda/ccache/netlist/schematic-connection.go
+share/lepton-eda/ccache/netlist/schematic-port.go
+share/lepton-eda/ccache/netlist/schematic.go
+share/lepton-eda/ccache/netlist/schematic/toplevel.go
+share/lepton-eda/ccache/netlist/subschematic-connection.go
+share/lepton-eda/ccache/netlist/subschematic.go
+share/lepton-eda/ccache/netlist/verbose.go
+share/lepton-eda/ccache/pcb.go
+share/lepton-eda/ccache/schematic/
+share/lepton-eda/ccache/schematic/action.go
+share/lepton-eda/ccache/schematic/attrib.go
+share/lepton-eda/ccache/schematic/builtins.go
+share/lepton-eda/ccache/schematic/core/
+share/lepton-eda/ccache/schematic/core/gettext.go
+share/lepton-eda/ccache/schematic/dialog.go
+share/lepton-eda/ccache/schematic/doc.go
+share/lepton-eda/ccache/schematic/ffi/
+share/lepton-eda/ccache/schematic/ffi.go
+share/lepton-eda/ccache/schematic/ffi/gobject.go
+share/lepton-eda/ccache/schematic/ffi/gtk.go
+share/lepton-eda/ccache/schematic/gui/
+share/lepton-eda/ccache/schematic/gui/keymap.go
+share/lepton-eda/ccache/schematic/gui/stroke.go
+share/lepton-eda/ccache/schematic/hook.go
+share/lepton-eda/ccache/schematic/keymap.go
+share/lepton-eda/ccache/schematic/menu.go
+share/lepton-eda/ccache/schematic/netlist.go
+share/lepton-eda/ccache/schematic/precompile.go
+share/lepton-eda/ccache/schematic/repl.go
+share/lepton-eda/ccache/schematic/selection.go
+share/lepton-eda/ccache/schematic/symbol/
+share/lepton-eda/ccache/schematic/symbol/check.go
+share/lepton-eda/ccache/schematic/undo.go
+share/lepton-eda/ccache/schematic/util.go
+share/lepton-eda/ccache/schematic/window.go
+share/lepton-eda/ccache/symbol/
+share/lepton-eda/ccache/symbol/blame.go
+share/lepton-eda/ccache/symbol/check/
+share/lepton-eda/ccache/symbol/check.go
+share/lepton-eda/ccache/symbol/check/alignment.go
+share/lepton-eda/ccache/symbol/check/arc.go
+share/lepton-eda/ccache/symbol/check/attrib.go
+share/lepton-eda/ccache/symbol/check/box.go
+share/lepton-eda/ccache/symbol/check/bus.go
+share/lepton-eda/ccache/symbol/check/circle.go
+share/lepton-eda/ccache/symbol/check/component.go
+share/lepton-eda/ccache/symbol/check/connection.go
+share/lepton-eda/ccache/symbol/check/duplicate.go
+share/lepton-eda/ccache/symbol/check/entity-pin.go
+share/lepton-eda/ccache/symbol/check/forbidden.go
+share/lepton-eda/ccache/symbol/check/line.go
+share/lepton-eda/ccache/symbol/check/log.go
+share/lepton-eda/ccache/symbol/check/net-attrib.go
+share/lepton-eda/ccache/symbol/check/net.go
+share/lepton-eda/ccache/symbol/check/obsolete.go
+share/lepton-eda/ccache/symbol/check/path.go
+share/lepton-eda/ccache/symbol/check/picture.go
+share/lepton-eda/ccache/symbol/check/pin-attrib.go
+share/lepton-eda/ccache/symbol/check/pin.go
+share/lepton-eda/ccache/symbol/check/primitive.go
+share/lepton-eda/ccache/symbol/check/slot.go
+share/lepton-eda/ccache/symbol/check/text.go
+share/lepton-eda/ccache/symcheck/
+share/lepton-eda/ccache/symcheck/check.go
+share/lepton-eda/ccache/symcheck/option.go
+share/lepton-eda/ccache/symcheck/report.go
 share/lepton-eda/geda-system.conf
 share/lepton-eda/gschem-colormap-bw
 share/lepton-eda/gschem-colormap-darkbg
@@ -975,8 +1134,6 @@ share/lepton-eda/lepton-system.conf
 share/lepton-eda/print-colormap-darkbg
 share/lepton-eda/print-colormap-lightbg
 share/lepton-eda/scheme/
-share/lepton-eda/scheme/auto-place-attribs.scm
-share/lepton-eda/scheme/auto-place-netname.scm
 share/lepton-eda/scheme/auto-refdes.scm
 share/lepton-eda/scheme/auto-uref.scm
 share/lepton-eda/scheme/autoload/
@@ -1031,7 +1188,6 @@ share/lepton-eda/scheme/conf/schematic/d
 share/lepton-eda/scheme/conf/schematic/keys.scm
 share/lepton-eda/scheme/conf/schematic/menu.scm
 share/lepton-eda/scheme/conf/schematic/stroke.scm
-share/lepton-eda/scheme/default-attrib-positions.scm
 share/lepton-eda/scheme/geda/
 share/lepton-eda/scheme/geda-deprecated-config.scm
 share/lepton-eda/scheme/geda.scm
@@ -1058,15 +1214,16 @@ share/lepton-eda/scheme/gschem/util.scm
 share/lepton-eda/scheme/gschem/window.scm
 share/lepton-eda/scheme/lepton/
 share/lepton-eda/scheme/lepton/attrib.scm
+share/lepton-eda/scheme/lepton/autoplace.scm
 share/lepton-eda/scheme/lepton/color-map.scm
 share/lepton-eda/scheme/lepton/config.scm
-share/lepton-eda/scheme/lepton/core/
-share/lepton-eda/scheme/lepton/core/gettext.scm
 share/lepton-eda/scheme/lepton/eval.scm
 share/lepton-eda/scheme/lepton/ffi/
 share/lepton-eda/scheme/lepton/ffi.scm
 share/lepton-eda/scheme/lepton/ffi/lib.scm
 share/lepton-eda/scheme/lepton/file-system.scm
+share/lepton-eda/scheme/lepton/gerror.scm
+share/lepton-eda/scheme/lepton/gettext.scm
 share/lepton-eda/scheme/lepton/legacy-config/
 share/lepton-eda/scheme/lepton/legacy-config.scm
 share/lepton-eda/scheme/lepton/legacy-config/keylist.scm
@@ -1079,6 +1236,7 @@ share/lepton-eda/scheme/lepton/m4.scm
 share/lepton-eda/scheme/lepton/object/
 share/lepton-eda/scheme/lepton/object.scm
 share/lepton-eda/scheme/lepton/object/foreign.scm
+share/lepton-eda/scheme/lepton/object/text.scm
 share/lepton-eda/scheme/lepton/object/type.scm
 share/lepton-eda/scheme/lepton/option.scm
 share/lepton-eda/scheme/lepton/os.scm
@@ -1088,6 +1246,7 @@ share/lepton-eda/scheme/lepton/page/fore
 share/lepton-eda/scheme/lepton/rc.scm
 share/lepton-eda/scheme/lepton/repl.scm
 share/lepton-eda/scheme/lepton/srfi-37.scm
+share/lepton-eda/scheme/lepton/toplevel.scm
 share/lepton-eda/scheme/lepton/version.scm
 share/lepton-eda/scheme/netlist/
 share/lepton-eda/scheme/netlist.scm
@@ -1126,11 +1285,11 @@ share/lepton-eda/scheme/schematic/builti
 share/lepton-eda/scheme/schematic/core/
 share/lepton-eda/scheme/schematic/core/gettext.scm
 share/lepton-eda/scheme/schematic/dialog.scm
+share/lepton-eda/scheme/schematic/doc.scm
 share/lepton-eda/scheme/schematic/ffi/
 share/lepton-eda/scheme/schematic/ffi.scm
 share/lepton-eda/scheme/schematic/ffi/gobject.scm
 share/lepton-eda/scheme/schematic/ffi/gtk.scm
-share/lepton-eda/scheme/schematic/gschemdoc.scm
 share/lepton-eda/scheme/schematic/gui/
 share/lepton-eda/scheme/schematic/gui/keymap.scm
 share/lepton-eda/scheme/schematic/gui/stroke.scm
